### ISSUE TITLE
Fatal Exception: java.lang.RuntimeException: Unable to start receiver com.urbanairship.gimbal.AirshipReadyReceiver: java.lang.IllegalArgumentException: Error: requested job be persisted without holding RECEIVE_BOOT_COMPLETED permission.

### ISSUE THREAD

**Issue by harishrpatel** [urbanairship/android-gimbal-adapter#22]
Started seeing this crash with Airship v14.0.3 and Gimbal Adapter v7.0.0.
OS: Android 10
Crash occurs during app launch (mostly seen on Instant App launch - We use in app automation in instant app).

Should the library AndroidManifest.xml have RECEIVE_BOOT_COMPLETED permission declared?

`Fatal Exception: java.lang.RuntimeException: Unable to start receiver com.urbanairship.gimbal.AirshipReadyReceiver: java.lang.IllegalArgumentException: Error: requested job be persisted without holding RECEIVE_BOOT_COMPLETED permission.
       at android.app.ActivityThread.handleReceiver(ActivityThread.java:3914)
       at android.app.ActivityThread.access$1700(ActivityThread.java:231)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1963)
       at android.os.Handler.dispatchMessage(Handler.java:107)
       at android.os.Looper.loop(Looper.java:214)
       at android.app.ActivityThread.main(ActivityThread.java:7682)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:516)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)`

**Comment by harishrpatel** [comment]
Here is another stack trace (seems to happen in Gimbal.start() call):
`Caused by java.lang.IllegalArgumentException: Error: requested job be persisted without holding RECEIVE_BOOT_COMPLETED permission.
       at android.os.Parcel.createException(Parcel.java:2075)
       at android.os.Parcel.readException(Parcel.java:2039)
       at android.os.Parcel.readException(Parcel.java:1987)
       at android.app.job.IJobScheduler$Stub$Proxy.schedule(IJobScheduler.java:308)
       at android.app.JobSchedulerImpl.schedule(JobSchedulerImpl.java:43)
       at com.gimbal.android.jobs.d.a(SourceFile:29)
       at com.gimbal.android.jobs.c.i(SourceFile:1)
       at com.gimbal.android.jobs.c.m(SourceFile:12)
       at com.gimbal.android.jobs.c.j(SourceFile:6)
       at f7.b.c(SourceFile:80)
       at f7.d.b(SourceFile:1)
       at f7.d.a(SourceFile:1)
       at com.gimbal.android.e.c(SourceFile:1)
       at com.urbanairship.gimbal.a.c(SourceFile:2)
       at com.xyz.MyApplication.onCreate(SourceFile:6)
       at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1197)
       at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6647)
       at android.app.ActivityThread.access$1600(ActivityThread.java:231)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1952)
       at android.os.Handler.dispatchMessage(Handler.java:107)
       at android.os.Looper.loop(Looper.java:214)
       at android.app.ActivityThread.main(ActivityThread.java:7682)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:516)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)`

**Comment by rlepinski** [comment]
@harishrpatel I am not sure if this is originating from Airship or Gimbal. We only try to schedule persistent jobs if we are rescheduling a push that failed to build the notification but we protect calls to it with a permission check:

```
        if (jobInfo.isPersistent() && ManifestUtils.isPermissionGranted(Manifest.permission.RECEIVE_BOOT_COMPLETED)) {
            builder.setPersisted(true);
        }
```

And we catch exceptions around job scheduling, so I am thinking its probably a Gimbal issue. Could you deobfuscate that stack trace for us? If it is Gimbal, we will have to start a support request through email.

**Comment by rlepinski** [comment]
@harishrpatel We are 2 versions behind on Gimbal, but I dont see anything about this issue in the changelog. The Gimbal SDK already defines that permission: `<uses-permission android:name="android.permission.RECEIVE_BOOT_COMPLETED" />`. Could you check your merged manifest as well as the output of `./gradlew app:dependencies` to see what version of Gimbal SDK you have?

**Comment by harishrpatel** [comment]
@rlepinski I sent you an email with deobfuscated stack trace.
- We have had the RECEIVE_BOOT_COMPLETED permission in our app's manifest for 3 years, but apparently not helping.
- Here is gradle info:
+--- com.urbanairship.android:gimbal-adapter:7.0.0
|    +--- com.gimbal.android.v4:gimbal-slf4j-impl:4.2.1
|    |    \--- org.slf4j:slf4j-api:1.7.13
|    +--- com.gimbal.android.v4:gimbal-sdk:4.2.1
|    \--- com.urbanairship.android:urbanairship-core:13.1.0 -> 14.0.3 (*)

**Comment by rlepinski** [comment]
@harishrpatel Fixed in the 7.1.0 with the new Gimbal SDK 4.5.0
